pkgs/development/web/nodejs/nodejs.nix

```diff
@@ -16,35 +15,110 @@
 let
   inherit (darwin.apple_sdk.frameworks) CoreServices ApplicationServices;
 
-  isCross = stdenv.hostPlatform != stdenv.buildPlatform;
-
   majorVersion = lib.versions.major version;
   minorVersion = lib.versions.minor version;
 
   pname = if enableNpm then "nodejs" else "nodejs-slim";
 
+  canExecute = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
+  emulator = stdenv.hostPlatform.emulator buildPackages;
+
+  # See valid_os and valid_arch in configure.py.
+  destOS =
+    let
+      platform = stdenv.hostPlatform;
+    in
+    if platform.isiOS then
+      "ios"
+    else if platform.isAndroid then
+      "android"
+    else if platform.isWindows then
+      "win"
+    else if platform.isDarwin then
+      "mac"
+    else if platform.isLinux then
+      "linux"
+    else if platform.isOpenBSD then
+      "openbsd"
+    else if platform.isFreeBSD then
+      "freebsd"
+    else
+      throw "unsupported os ${platform.uname.system}";
+  destCPU =
+    let
+      platform = stdenv.hostPlatform;
+    in
+    if platform.isAarch then
+      "arm" + lib.optionalString platform.is64bit "64"
+    else if platform.isMips32 then
+      "mips" + lib.optionalString platform.isLittleEndian "le"
+    else if platform.isMips64 && platform.isLittleEndian then
+      "mips64el"
+    else if platform.isPower then
+      "ppc" + lib.optionalString platform.is64bit "64"
+    else if platform.isx86_64 then
+      "x64"
+    else if platform.isx86_32 then
+      "ia32"
+    else if platform.isS390x then
+      "s390x"
+    else if platform.isRiscV64 then
+      "riscv64"
+    else if platform.isLoongArch64 then
+      "loong64"
+    else
+      throw "unsupported cpu ${platform.uname.processor}";
```

Comment:
This is not an idiomatic Nixpkgs pattern and should probably use some fallback for unknown platforms instead of throwing an error. I’m keeping it for now and will address this in subsequent code style PRs.

Reply:
It's fine to throw if we don't know the platform.